Ginger/GingerAutoPilot/APIModelLib/ImportParametersOptionalValues.cs

```diff
@@ -0,0 +1,268 @@
+        {
+            int UpdatedParametersCounter = 0;
+            foreach (AppModelParameter AMP in AAM.AppModelParameters)
+            {
+                string result = AMP.Path;
+                Tuple<string, string> tuple = new Tuple<string, string>(AMP.TagName, AMP.Path);
+                if (OptionalValuesPerParameterDict.ContainsKey(tuple))
+                {
+                    PopulateOptionalValuesByTuple(AMP, OptionalValuesPerParameterDict, tuple);
+                    if (APIConfigurationsDocumentParserBase.ParameterValuesUpdated)
+                    {
+                        UpdatedParametersCounter++;
+                        APIConfigurationsDocumentParserBase.ParameterValuesUpdated = false;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Update optional values only for selected parameter according to json file
+        /// </summary>
+        /// <param name="AAM"></param>
+        /// <param name="OptionalValuesPerParameterDict"></param>
+        /// <param name="SelectedParametersGridList"></param>
+        public void PopulateJSONOptionalValuesForAPIParameters(ApplicationModelBase AAM, Dictionary<Tuple<string, string>, List<string>> OptionalValuesPerParameterDict, List<AppModelParameter> SelectedParametersGridList)
+        {
+            int UpdatedParametersCounter = 0;
+            foreach (var tuple in SelectedParametersGridList.Zip(AAM.AppModelParameters, (x, y) => (x, y)))
+            {
+                if (tuple.x.RequiredAsInput)//selected
+                {
+                    Tuple<string, string> tupleKey = new Tuple<string, string>(tuple.y.TagName, tuple.y.Path);
+                    if (OptionalValuesPerParameterDict.ContainsKey(tupleKey))
+                    {
+                        PopulateOptionalValuesByTuple(tuple.y, OptionalValuesPerParameterDict, tupleKey);
+                        if (APIConfigurationsDocumentParserBase.ParameterValuesUpdated)
+                        {
+                            UpdatedParametersCounter++;
+                            APIConfigurationsDocumentParserBase.ParameterValuesUpdated = false;
+                        }
+                    }
+                }
+            }
+        }
+        #endregion
+
+        private void PopulateOptionalValuesByTuple(AppModelParameter AMP, Dictionary<Tuple<string, string>, List<string>> OptionalValuesPerParameterDict, Tuple<string, string> tuple)
+        {
+            foreach (string Value in OptionalValuesPerParameterDict[tuple])
+            {
+                OptionalValue OptionalValueExist = AMP.OptionalValuesList.Where(x => x.Value == Value).FirstOrDefault();
+                if (OptionalValueExist == null)
+                {
+                    OptionalValue OptionalValue = new OptionalValue() { Value = Value };
+                    if (!string.IsNullOrEmpty(Value))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4464228)